src/plugins/intel_gpu/src/kernel_selector/kernels/resample/resample_kernel_pil_ref.cpp

```diff
@@ -52,18 +52,38 @@ Tensor::Dim ExtractDim(const DataTensor& tensor, InterpolateAxis axis) {
     return tensor.Extract(tensor.GetLayout(), Convert(axis), tensor.GetDims());
 }
 
+size_t getInputHorizontalSize(const resample_params& params, bool with_pad = false) {
+    auto inputHorizontalSize = ExtractDim(params.inputs[0], params.axes[eHorizontal]).v;
+    if (with_pad) {
+        inputHorizontalSize += params.pads_begin[ConvertToNCHW(params.axes[eHorizontal])];
+        inputHorizontalSize += params.pads_end[ConvertToNCHW(params.axes[eHorizontal])];
+    }
+    return inputHorizontalSize;
+}
+
+size_t getInputVerticalSize(const resample_params& params, bool with_pad = false) {
+    auto inputVerticalSize = ExtractDim(params.inputs[0], params.axes[eVertical]).v;
+    if (with_pad) {
+        inputVerticalSize += params.pads_begin[ConvertToNCHW(params.axes[eVertical])];
+        inputVerticalSize += params.pads_end[ConvertToNCHW(params.axes[eVertical])];
+    }
+    return inputVerticalSize;
+}
```

Comment:
is there a case where `with_pad = false` for `getInputHorizontalSize` or `getInputVerticalSize`? If not, I think we can simplify this function.